ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/TlsConfiguration.scala

```diff
@@ -51,24 +42,65 @@ final case class TlsConfiguration(
 
   /** If enabled and all required fields are present, it returns an SslContext suitable for server usage */
   def server: Option[SslContext] =
-    if (enabled)
-      Some(
-        GrpcSslContexts
-          .forServer(
-            keyCertChainFileOrFail,
-            keyFileOrFail,
-          )
-          .trustManager(trustCertCollectionFile.orNull)
-          .clientAuth(clientAuth)
-          .protocols(if (protocols.nonEmpty) protocols.asJava else null)
-          .build
-      )
-    else None
+    if (enabled) {
+      scala.util.Using.resources(
+        keyCertChainInputStreamOrFail,
+        keyInputStreamOrFail,
+      ) { (keyCertChain: InputStream, key: InputStream) =>
+        Some(
+          GrpcSslContexts
+            .forServer(
+              keyCertChain,
+              key,
+            )
+            .trustManager(trustCertCollectionFile.orNull)
+            .clientAuth(clientAuth)
+            .protocols(if (protocols.nonEmpty) protocols.asJava else null)
+            .build
+        )
+
+      }
+    } else None
 
   /** This is a side-effecting method. It modifies JVM TLS properties according to the TLS configuration. */
   def setJvmTlsProperties(): Unit =
     if (enabled && enableCertRevocationChecking) OcspProperties.enableOcsp()
 
+  private[tls] def keyInputStreamOrFail: InputStream = {
```

Comment:
Why did you refactor these methods to return a Stream instead of a File? Is it in order to properly release resources in case of failures? If yes, then please have a look at our `Resource` and `ResourceOwner` classes. They allow you to compose the acquisition of resources with proper cleanup. Search for `ResourceOwner.forCloseable`, for example:

```scala
// This defines how to acquire the streams
inputStreamOwner = for {
  stream1 <- ResourceOwner.forCloseable(() => openSomeStream())
  stream2 <- ResourceOwner.forCloseable(() => openSomeStream())
} yield (stream1, stream2)

// This will open the streams, call foo(), and close the streams, properly cleaning up in case of failures.
inputStreamOwner.use { case (stream1, stream2) => foo(stream1, stream2)}
```

Reply:
The reasons for the input streams are:
1) `java.io.File` was not an option for storing the decrypted private key because in this story we actually want to avoid having it in plaintext on the filesystem.
2) In `TlsConfiguration` we create [1] an instance of `io.grpc.netty.GrpcSslContexts` which accepts `File`s or `InputStream`s. So I went with the latter, especially that I could then abstract over reading from a file and a byte array.

[1]
```
GrpcSslContexts
            .forServer(
              keyCertChain,
              key,
            )
```      
https://github.com/digital-asset/daml/blob/pbatko/dpp-418-protect-tls-keys/ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/TlsConfiguration.scala#L52

Reply:
I see, no need to use ResourceOwner then.